### PR TITLE
C2 rev cotton

### PR DESCRIPTION
**Pull request by Robinlovelace**
Pls don't merge this yet - want to keep this open for comments/suggestions/questions from you and perhaps others (e.g. @richierocks - many thanks for the detailed review!).

### CODE REVIEW

File: 02-set-up.Rmd [hunk 1]
```diff
@@ -177,8 +177,15 @@ identical(r1, r2)
...
+This applies to an even greater extent to packages, the behaviour of which can change substantially from one release to the next.
+Depending on context, you may want to your R installation and packages up-to-date, or 'pin' the version of R and packages to specific versions, to ensure stability.
+For most desktop users, keeping packages up-to-date is desirable: new code tends to be more efficient, intuitive, robust and feature rich. This section explains how.
+
+```{block, packrat, style='rmdtip'}
+Previous R versions can be installed from CRAN's archive or previous R releases. The binary versions for all OSs can be found at [cran.r-project.org/bin/](https://cran.r-project.org/bin/). To download binary versions for Ubuntu 'Wily', for example, see [cran.r-project.org/bin/linux/ubuntu/wily/](https://cran.r-project.org/bin/linux/ubuntu/wily/). To 'pin' specific versions of R packages you can use the **packrat** package. For more on pinning R versions and R packages see articles on RStudio's website [Using-Different-Versions-of-R](https://support.rstudio.com/hc/en-us/articles/200486138-Using-Different-Versions-of-R) and [rstudio.github.io/packrat/](https://rstudio.github.io/packrat/).
```
Comment (csgillespie) [inline 02-set-up.Rmd]: Is pin a technical packrat term?

File: 02-set-up.Rmd [hunk 2]
```diff
@@ -17,7 +17,8 @@ Starting with the basics and moving to progressively more advanced topics, we ex
 R version, startup files and IDE can make your R work faster (though IDE could be seen as basic need for efficient programming).
 Ensuring correct configuration of these elements will have knock-on benefits in many aspects of your R workflow. That's why we cover them at this early stage (hardware, the other fundamental consideration, is covered in the next chapter).
 By the end of this chapter you should understand how to set-up your computer
-and R installation for optimal computational and programmer efficiency.
+and R installation (skip to section \@ref(r-version) if R is not already installed on your computer)
```
Comment (csgillespie) [inline 02-set-up.Rmd]: Do we need to tell people to install R if they are reading this book? In theory they have already done Garett's course

File: 02-set-up.Rmd [hunk 3]
```diff
@@ -13,11 +13,11 @@ its components work in harmony, it is well-serviced,
 and it is fast.
 This chapter describes the software decisions that will enable
 a productive workflow. We explore how the operating system,
-R versions and settings, configuration of your R editor, and
-other settings can make your R work faster.
-The next chapter deals with hardware.
-By the end of this chapter you should understand how to optimize your computer for efficient R programming.
-That includes consideration of the following topics:
+R version, startup files and IDE can make your R work faster.
```
Comment (csgillespie) [inline 02-set-up.Rmd]: Just checking. Has IDE been defined?

File: efficient-r-feedback-chapter2-rcotton.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
+## General points
+
+I think it's worth making an effort between distinguishing the bits of book that are how-to tips, and the bits that are explaining underlying concepts.
+
+You also need to sell some ideas harder.  For example, describe a problem that some your advice then solves.  For example, I'm a lousy typist, and autocomplete means fewer typos.  Or, R has a lots of functions, and remembering their names can be a nightmare.  This is where `apropos` and `??` come in useful.
```
Comment (csgillespie) [inline efficient-r-feedback-chapter2-rcotton.md]: Use a rmdnote to mention that lots of functions is 100,000's http://r4stats.com/2016/04/19/rs-growth-continues-to-accelerate/

File: 02-set-up.Rmd [hunk 5]
```diff
@@ -46,9 +46,9 @@ This is partly facilitated by CRAN tests which ensure that
 R packages work on all OSs mentioned above.
 There are some operating system-specific quirks that may influence the choice
 of OS and how it is set-up for R programming in the long-term.
-Basic system information can be queried from within R using `Sys.info()`, as illustrated below for a selection of the functions output elements:
+Basic system information can be queried from within R using `Sys.info()`, as illustrated below for a selection its output:
 
-```{r}
+```{r, eval=FALSE}
```
Comment (csgillespie) [inline 02-set-up.Rmd]: As an aside, I tend to use `eval=TRUE, results="hide"` to avoid typos within R code.

File: 02-set-up.Rmd [hunk 6]
```diff
@@ -105,7 +105,8 @@ our advice is usually to stick to whatever OS you are most comfortable with.
 Some packages (e.g. those that must be compiled and that depend on external libraries) are best installed at the operating system level (i.e. not using `install.packages`) on Linux systems.
 On Debian-based operating systems such as Ubuntu, these are named with the prefix `r-cran-` (see [Section 2.4](#deps)).
 
-Regardless of your operating system, it is good practice to track how system resources (primarily CPU and RAM use) respond when running time-consuming tasks.
+Regardless of your operating system, it is good practice to track how system resources (primarily CPU and RAM use) respond when running time-consuming or RAM-intensive tasks.
+If you only process small datasets system monitoring may not be necessary but when handling datasets at the limits of your computer's resources, it can be a useful tool for identifying bottlenecks, such as when you are running low on RAM.
 Alongside R profiling functions such as `profvis` (see Section __XXX__), system monitoring can help identify performance bottlenecks and opportunities for making tasks run faster.
```
Comment (csgillespie) [inline 02-set-up.Rmd]: `profvis` -> **profvis**

File: 02-set-up.Rmd (hunk 1 above)
Comment (Robinlovelace) [inline 02-set-up.Rmd]: It's a technical term (e.g. see [here](https://help.ubuntu.com/community/PinningHowto)), but not one that's [used in packrat](https://github.com/rstudio/packrat/search?utf8=%E2%9C%93&q=pin).

File: 02-set-up.Rmd (hunk 6 above)
Comment (Robinlovelace) [inline 02-set-up.Rmd]: Fixed - thanks for flagging that.

**Comment by csgillespie** [comment]
@Robinlovelace  Should I merge?

**Comment by Robinlovelace** [comment]
@csgillespie you can now - just completed the review.

**Comment by Robinlovelace** [comment]
Ps check out my new (v. simple but gets the message across) figure 2.4.

**Pull request merged**